onset_maproom/onset_indices.py

```diff
@@ -35,6 +36,62 @@ def onset_indices(
     onset_ids = np.unique(first_rainy[onset_ids] + onset_ids)
     return onset_ids
 
+def onset_index(
+    data, data_flag, wlength,      wthresh,       flagged_thresh, fwlength, false_search, coord_index
+):
+    """Finds the first window of size wlength where data exceeds wthresh,
+    with at least flagged_thresh count of flagged data (by data_flag),
+    not followed by a false_window of size fwlength of unflagged data, for the following false_search indices
+    returns the index of first flagged data in that window
+    (how much wood would a woodchuck chuck if a woodchuck could chuck wood?)
+    """
```

Comment:
Personally, I don't believe there is another use case for this. Even cessation date, I don't think it can apply. It can inspire a cessation date function and other certainly though. Assuming input is daily data will greatly simplify things. In particular the last step to find the first wet day within the first valid wet spell. I could use arg/idxmin to get the relative position in the group and then use that as an offset to the final result that finds the position of the last day of the wet spell. I actually did that originally until I realized I did assume daily data. What I did didn't work on indices because I ran into cases where idxmin and argmin respectively gave a date and a NaN when applied to the same input. And once idxmin is used, we are in the coordinates domain and no more in indices domain.
Anyway, my point is that generalizing brings constrains that can seriously change the philosophy of the code so I'd suggest either we stick to general, or we go for an onset_date function that assume daily rainfall as input.

Reply:
For the little history about "flag." In Ingrid, flag functions return 1 or 0 based on a condition. And in many datasets we are used to manipulate, a flag variable is often a quality control key that can be more subtle than binary, e.g. 'observed', 'estimated', 'filled', interpolated', or more specialized term. So probably flag should not be connoted "bad."

Reply:
The groupby function (your other PR) already assumes daily data, so no point making this one more general yet, especially since of the two components, this one is less likely to be reusable.